From ab6e711370140aaf594db80d82440fac3b038837 Mon Sep 17 00:00:00 2001
From: kelson8 <kelson@kelsoncraft.net>
Date: Mon, 17 Aug 2026 18:10:59 -0400
Subject: [PATCH 071/161] feat: Add get player health into lua

* Add pushIntToLua and pushFloatToLua functions.
* Copied changes from miami-tcp-test branch.
---
 .../lua_scripts/kcnet-keybind-events.lua         |  3 +++
 src/extras/lua/lua_functions.cpp                 | 16 ++++++++++++++++
 src/extras/lua/lua_functions.h                   |  3 +++
 src/extras/lua/lua_player.cpp                    | 14 +++++++++++++-
 src/extras/lua/lua_player.h                      |  1 +
 src/extras/lua/lua_test.cpp                      |  1 +
 6 files changed, 37 insertions(+), 1 deletion(-)

diff --git a/gamefiles/ViceExtended/lua_scripts/kcnet-keybind-events.lua b/gamefiles/ViceExtended/lua_scripts/kcnet-keybind-events.lua
index af845e02..75bb233b 100644
--- a/gamefiles/ViceExtended/lua_scripts/kcnet-keybind-events.lua
+++ b/gamefiles/ViceExtended/lua_scripts/kcnet-keybind-events.lua
@@ -134,3 +134,6 @@ end
 
 -- This won't work until I fix it.
 -- change_clothes("SoNny ForellI")
+
+-- Get the players health
+-- print("[ReVC]: Players health: " .. player_health())
diff --git a/src/extras/lua/lua_functions.cpp b/src/extras/lua/lua_functions.cpp
index 6d410543..10e7123d 100644
--- a/src/extras/lua/lua_functions.cpp
+++ b/src/extras/lua/lua_functions.cpp
@@ -97,6 +97,22 @@ pushBooleanToLua(lua_State *L, bool value)
 	lua_pushboolean(L, value);
 }
 
+/**
+* Push a int to lua.
+*/
+void pushIntToLua(lua_State *L, int value)
+{
+	lua_pushnumber(L, value);
+}
+	
+/**
+* Push a float to lua.
+*/
+void pushFloatToLua(lua_State *L, float value)
+{
+	lua_pushnumber(L, value);
+}
+
 // Check lua globals, made this into a function
 // Moved out of lua_test.cpp.
 bool
diff --git a/src/extras/lua/lua_functions.h b/src/extras/lua/lua_functions.h
index 78a60499..5bf58007 100644
--- a/src/extras/lua/lua_functions.h
+++ b/src/extras/lua/lua_functions.h
@@ -14,6 +14,9 @@ extern "C" {
 
 	//------------- Functions -----------//
 	extern void pushBooleanToLua(lua_State *L, bool value);
+	extern void pushIntToLua(lua_State *L, int value);
+	extern void pushFloatToLua(lua_State *L, float value);
+
 	extern bool CheckLua(lua_State *L, int r);
 
 #ifdef __cplusplus
diff --git a/src/extras/lua/lua_player.cpp b/src/extras/lua/lua_player.cpp
index 118523be..24505bc4 100644
--- a/src/extras/lua/lua_player.cpp
+++ b/src/extras/lua/lua_player.cpp
@@ -24,8 +24,8 @@
 // VC Specific
 #include "Hud.h"
 #include "Radar.h"
-#include "Text.h"
 #include "Restart.h"
+#include "Text.h"
 
 #include "Pad.h"
 
@@ -143,6 +143,18 @@ HealPlayerLua(lua_State *L)
 	return 1;
 }
 
+/**
+ * Get the players health with Lua.
+ */
+int
+GetPlayerHealthLua(lua_State *L)
+{
+	float playerHealth = FindPlayerPed()->m_fHealth;
+	if(playerHealth) { pushFloatToLua(L, playerHealth); }
+
+	return 1;
+}
+
 //------- Vehicle --------//
 
 /// <summary>
diff --git a/src/extras/lua/lua_player.h b/src/extras/lua/lua_player.h
index 7319537c..5b3260e1 100644
--- a/src/extras/lua/lua_player.h
+++ b/src/extras/lua/lua_player.h
@@ -25,6 +25,7 @@ extern "C" {
 	extern int TogglePlayerControlLua(lua_State *L);
 	extern int HealPlayerLua(lua_State *L);
 
+	extern int GetPlayerHealthLua(lua_State *L);
 	// 
 	extern int enableNeverWantedLua(lua_State *L);
 	extern int enablePlayerInfiniteHealthLua(lua_State *L);
diff --git a/src/extras/lua/lua_test.cpp b/src/extras/lua/lua_test.cpp
index a0d1688a..15157736 100644
--- a/src/extras/lua/lua_test.cpp
+++ b/src/extras/lua/lua_test.cpp
@@ -368,6 +368,7 @@ LuaTest::RegisterLuaFunctions(lua_State *L)
 	// lua_register(LuaTest::GetInstance().GetLuaState(), "set_position", SetPlayerPositionLua);
 
 	lua_register(L, "heal_player", HealPlayerLua);
+	lua_register(L, "player_health", GetPlayerHealthLua);
 
 	// This works
 	// TODO Get CVector for respawn from file.
-- 
2.43.0

